filestore/src/main/java/io/syndesis/filestore/FileStoreException.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.filestore;
+
+/**
+ * These runtime exceptions are thrown by the methods of the FileStore
+ */
+public class FileStoreException extends RuntimeException {
+
+    public FileStoreException(String message) {
+        super(message);
+    }
+
+    public FileStoreException(String message, Throwable cause) {
+        super(message, cause);
+    }
+}
```

Comment:
Can you add a launderThrowable method and use it? Like https://github.com/syndesisio/syndesis-rest/blob/master/core/src/main/java/io/syndesis/core/SyndesisServerException.java#L43-L55

Reply:
I can add a launderThrowable but the filestore exception is always used to add context information (e.g. the path being written) when a low level exception occurs. Since DBI exceptions (the low level exceptions) are always runtime exceptions, the message would be discarded, the runtime exception rethrown as is.. and this exception would not be used..

Reply:
Yes ignore my comment, you actually want it to be a runtime exception.